tests/generative/variable_strategies.py

```diff
@@ -182,19 +184,70 @@ def variable(patient_tables, event_tables, schema, int_values, bool_values):
     add = qm_builds(Function.Add, series, series)
     subtract = qm_builds(Function.Subtract, series, series)
 
+    assert_complete_coverage()
+
     # Variables must be single values which have been reduced to the patient level. We also need to ensure that they
     # contains nodes which actually access the database so that the query engine can calculate a patient universe.
     return one_row_per_patient_series.filter(uses_the_database)
 
 
 # A specialized version of st.builds() which cleanly rejects invalid Query Model objects.
-@st.composite
-def qm_builds(draw, type_, *arg_strategies):
-    args = [draw(s) for s in arg_strategies]
-    try:
-        return type_(*args)
-    except ValidationError:
-        raise hypothesis.errors.UnsatisfiedAssumption
+# We also record the QM operations for which strategies have been created and then assert that
+# this includes all operations that exist, to act as a reminder to us to add new operations here
+# when they are added to the query model.
```

Comment:
Oh great! I didn't intended to suggest that this ought to be done immediately, but I'm glad we've got this in place. I think it will be a useful guard against generative test rot.